pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -188,10 +238,67 @@ pub mod pallet {
 		/// Emitted when you can `end_batch_message()` but the packing process
 		/// was not started by `start_batch_message()`.
 		MessagePackingNotStarted,
+
+		/// Invalid routers.
+		InvalidRouters,
+
+		/// Unknown router.
+		UnknownRouter,
+
+		/// The router that sent the message is not the first one.
+		MessageExpectedFromFirstRouter,
+
+		/// The router that sent the proof should not be the first one.
+		ProofNotExpectedFromFirstRouter,
+
+		/// A message was expected instead of a proof.
+		ExpectedMessageType,
+
+		/// A message proof was expected instead of a message.
+		ExpectedMessageProofType,
+
+		/// Pending inbound entry not found.
+		PendingInboundEntryNotFound,
+
+		/// Message proof cannot be retrieved.
+		MessageProofRetrieval,
+
+		/// Recovery message not found.
+		RecoveryMessageNotFound,
+
+		/// Not enough routers are stored for a domain.
+		NotEnoughRoutersForDomain,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// Sets the router IDs used for a specific domain,
+		#[pallet::weight(T::WeightInfo::set_routers())]
+		#[pallet::call_index(0)]
```

Comment:
Another reason would be XCM right? Do we call this pallet via XCM in any of the multi-chain setups?

Reply:
> Another reason would be XCM right? 
Yes, XCM is the main reason because the indices enable access (i.e. extrinsics, assets etc) to remote chains without commiting to specific types.

>  Do we call this pallet via XCM in any of the multi-chain setups?
No we don't and this particular extrinsic, as well as the deprecated one, require root origin which cannot be met via XCM.

Reply:
OK, I'm glad we get away with it this time.